proc_CPMG.py

```diff
@@ -103,10 +72,12 @@ def print_fun(x, f, accepted):
     data = s['t2':(-200,200)].sum('t2')
     #data = s['t2':(0,200)].sum('t2')
     fl.next('Echo decay')
-    x = tE_axis
+
+    x = s.getaxis('nEchoes')
```

Comment:
I believe it's just grabbing the axis and calling it x now instead

Reply:
yes, but you don't even need this -- there's a plot command a couple lines down, and that should just take an nddata, rather than needing to break apart the axis + data.

Reply:
Okay so I looked at this again and the fit function uses the x axis as a variable in:
fitfunc = lambda p, x: p[0]*exp(-x*p[1])
    errfunc = lambda p_arg, x_arg, y_arg: fitfunc(p_arg, x_arg) - y_arg
    p0 = [0.1,100.0,-3.0]
    p1, success = leastsq(errfunc, p0[:], args=(x, ydata))
    assert success == 1, "Fit did not succeed"
    T2 = 1./p1[1]
    print(T2)
    x_fit = linspace(x.min(),x.max(),5000)
    fl.plot(x_fit, fitfunc(p1, x_fit),':', label='fit (T2 = %0.2f ms)'%(T2*1e3), human_units=False)
    xlabel('t (sec)')
    ylabel('Intensity')
    print("T2:",T2,"s")

Reply:
first, note you can do a code block with a triple backtick  this is actually hard to read, so I do that here:
```
fitfunc = lambda p, x: p[0]exp(-xp[1])
errfunc = lambda p_arg, x_arg, y_arg: fitfunc(p_arg, x_arg) - y_arg
p0 = [0.1,100.0,-3.0]
p1, success = leastsq(errfunc, p0[:], args=(x, ydata))
assert success == 1, "Fit did not succeed"
T2 = 1./p1[1]
print(T2)
x_fit = linspace(x.min(),x.max(),5000)
fl.plot(x_fit, fitfunc(p1, x_fit),':', label='fit (T2 = %0.2f ms)'%(T2*1e3), human_units=False)
xlabel('t (sec)')
ylabel('Intensity')
print("T2:",T2,"s")
```

Reply:
To reply, I would say please quickly recruit Alec.  First, ask him to generate a pull request that will merge in the most recent fitting capabilities.  Second, ask him if he can set it up so that you can pass the guess as a dictionary (so that you can pass the exact same guess that you are using here).  Finally, I would first yourself create a jupyter notebook (not checked in, just for your own playing around) with:
```
import sympy as s
s.init_printing()
```
you can then use `s.symbols(` to create symbols/variables that allow you to build the mathematical expression for your fitting function, and you can use the jupyter notebook to figure out the code to give you the function you want for an inversion recovery curve.

In parallel, once Alec generates the pull request, you can use the upgraded fitdata class to do this, so you don't need to pull apart the x axis this way (that's achieved with the `eval` method of `fitdata`).  When doing this, rather than trying to talk to Alec directly, as much as possible, please please get him to update the documentation, which is [here](https://jmfrancklab.github.io/pyspecdata/fitdata.html) so that you can just rewrite this code based on that documentation page

Reply:
discussed with alec and he will update me when fitfunction is pulled and documentation is updated